src/python/deepgnn/graph_engine/snark/decoders.py

```diff
@@ -1,220 +1,220 @@
-        for feature in node_features:
-            val = feature.split(":")
-            if len(val) != 2 or not val[0] or not val[1]:
-                feature_map.append(None)
-                continue
-
-            if val[0][0] == "i":
-                key = f"int{val[0][1:]}_feature"
-            elif val[0][0] == "u":
-                key = f"uint{val[0][1:]}_feature"
-            elif val[0][0] == "f":
-                key = f"float{val[0][1:]}_feature"
-            elif val[0][0] == "d":
-                key = "double_feature"
-            elif val[0][0] == "b":
-                key = "binary_feature"
-
-            if len(val) != 2 or not val[0] or not val[1]:
-                value = None
-            elif key != "binary_feature":
-                value = np.array(val[1].split(" "), dtype=self.convert_map[key])
-            else:
-                value = val[1]
-
-            feature_map.append(value)
-
-        return feature_map
-
-    def decode(self, line: str) -> Iterator[Tuple[int, int, int, float, list]]:
-        """Decode tsv based text line into a node and edge iterator.
-
-        This is a generator that yields a node then its outgoing edges in order.
-        Yield format is (node_id/src, -1/dst, type, weight, features).
-        Features being a list of dense features as ndarrays and sparse features as 2 tuples, coordinates and values.
-        """
-        assert line is not None and len(line) > 0
-
-        columns = next(csv.reader([line], delimiter="\t"))
-        # fill the missing columns with default empty string, so we
-        # don't need to check if the column exists.
-        for i in range(TsvDecoder.COLUMN_COUNT - len(columns)):
-            columns.append("")
-
-        # make sure the node id is valid
-        assert columns[0] is not None and len(columns[0]) > 0
-
-        node_id = int(columns[0])
-        node_type = int(columns[1]) if columns[1] else 0
-        node_weight = float(columns[2]) if columns[2] else 0.0
-        yield node_id, -1, node_type, node_weight, self._parse_feature_string(
-            columns[3]
-        )
-
-        if not columns[4]:
-            return
-
-        neighbors = columns[4].split("|")
-        for n in sorted(neighbors, key=lambda x: int(x.split(",")[1])):
-            neighbor_columns = n.split(",")
-            # make sure the destination id is valid
-            assert len(neighbor_columns) > 0 and len(neighbor_columns[0]) > 0
-            dst_id = int(neighbor_columns[0])
-
-            for i in range(TsvDecoder.NEIGHBOR_COLUMN_COUNT - len(neighbor_columns)):
-                neighbor_columns.append("")
-
-            dst_type = int(neighbor_columns[1]) if neighbor_columns[1] else 0
-            dst_weight = float(neighbor_columns[2]) if neighbor_columns[2] else 0.0
-            yield node_id, dst_id, dst_type, dst_weight, self._parse_feature_string(
-                neighbor_columns[3]
-            )
+# Copyright (c) Microsoft Corporation.

```

Comment:
Looks like you missed to clean line endings in this file, e.g. look at `edge_string_features` method.